### PR TITLE
Versioning2

### PR DESCRIPTION
**Pull request by Error323**
We got a file `version.inc` and a script `bumpversion.py` that can manipulate it. This should define a sane basis for our versioning needs.

### CODE REVIEW

**Comment by Error323** [comment]
The idea is now to flesh out bumpversion.py together such that it will suit our needs.

File: src/version.h [hunk 1]
```diff
@@ -2,20 +2,7 @@
 
 // Versioning is performed according to the standard at <https://semver.org/>
 // Our postfix consists of the 8 character git-sha1 id iff the commit is not
```
Comment (Tilps) [inline src/version.h]: This sentence looks obsolete, consider removing.

File: scripts/bumpversion.py [hunk 2]
```diff
@@ -1,48 +1,88 @@
...
+    with tempfile.NamedTemporaryFile(suffix='.md') as f:
+        f.write(init_msg.encode('utf-8'))
+        f.flush()
+        call([EDITOR, f.name])
+        f.seek(0)
+        return f.read().decode('utf-8')
+
+
+def update(major, minor, patch):
+    # this is a workflow suggestion:
```
Comment (Tilps) [inline scripts/bumpversion.py]: Maybe drop the workflow suggestion for now?

File: scripts/bumpversion.py [hunk 3]
```diff
@@ -1,48 +1,88 @@
...
+    call(["git", "commit", "-am", msg])
+
+
 def tag(major, minor, patch):
     version = "v{}.{}.{}".format(major, minor, patch)
-    subprocess.getoutput("git tag -m {} {}".format(version, version))
+    call(["git", "-m", version, version])
+
+
+def editor(init_msg):
```
Comment (Tilps) [inline scripts/bumpversion.py]: This seems like deadcode - I can see its potential use in future, but it doesn't look terribly complicated so maybe just drop for now?

**Pull request merged**